README.md

```diff
@@ -1,68 +1,256 @@
+## Terminology
 
-This eliminates one of the headaches of a reviewer: the minutes and hours spent waiting for tests to finish after code has been reviewed, to merge in the finished branch.
+* _Merging_: Merging a pull request involves many steps, including
+  automated CI tests, PR labeling, and the target branch update. Do not
+  confuse this general/imprecise term with the atomic action of merging
+  one git branch into another.
 
-It's my solution to my own [StackOverflow Question](http://stackoverflow.com/questions/40391344/automatically-merge-verified-and-tested-github-pull-requests/40398678#40398678)
+* _Core developer_: From the bot point of view, a core developer is a
+  GitHub user with write access to the configured repository (`config::repo`).
 
-Configuration
-=============
 
-EggTimer is meant to be run as a web server, which is then called
-by [GitHub's webhook framework](https://developer.github.com/webhooks/).
+## Which pull requests are eligible for merging?
 
-Customize config.js
--------------------
+A pull request is ready for merging if all of the following conditions
+are satisfied:
 
-Create a config.js, by cloning config-example.js:
+* The PR is open (TODO: Check that a PR closed right _after_ we got a
+  list of PRs will not be merged!)
+* The PR has GitHub 'mergeable' status. TODO: Document how a GitHub user
+  can tell whether a PR has a GitHub 'mergeable' status.
+* All the _required_ checks have succeeded on the (possibly stale) PR
+  branch. GitHub says "All checks have passed" next to a green check
+  mark. TODO: Show the right message screenshot here if possible. TODO:
+  What does GitHub say when an optional check has failed but all the
+  required ones have passed?
+* The PR is approved for merging (see below for voting rules).
+* The PR does not have an `S-merged` label.
 
-```
-cp config-example.js config.js
-```
+Satisfying the above conditions results in an attempt to merge the pull
+request (in FIFO order), but PR merging may still fail.
+
+
+## Pull request merging algorithm
+
+Each open pull request is processed according to the following
+algorithm:
+
+1. Reset the staging (a.k.a. "auto") branch to point at the last commit
+   of the PR target branch (usually "master").
+2. Squash-merge the PR branch into the staging branch, using PR title
+   (with an appended PR number) and the PR description as the commit
+   message. TODO: What happens to long paragraph lines often present in
+   the PR description? We should either reject PRs that have long lines
+   or, if it can be done safely, automatically wrap them using some
+   formatting library.
```

Comment:
It would be better(safer) to reject such PRs with long lines and PRs with GitHub markdown. I have read several guidelines about writing commit messages, all of them prescribe to keep to 72 characters per line.
Alternatively, we can try to automatically format these messages with some available tools: e.g., firstly [remove](https://github.com/stiang/remove-markdown) markdown and then [wrap](https://www.npmjs.com/package/word-wrap) the message. However, I have not tested these tools and don't know whether they are reliable.

Reply:
For now, I suggest disqualifying PRs that use excessively long lines and allowing/using everything else "as is". We will probably need a new label to mark/identify affected PRs (in addition to the merge-failed label that will also be applied).

The following items become TODOs for future work, in rough priority order:

1. Auto-convert markdown to plain text during merge, before checking for long lines. This will take care of most long lines, verbatim block quotes, and enumerated lists.
1. Provide a way to disable this check via a PR comment command. Some long lines should be admitted.
1. Provide a way to auto-convert the PR description via a PR comment command, with the original text posted as a comment for backup purposes.
1. Prohibit undesirable markdown while still allowing simple things like `_foo_` and `*bar*`.